contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1236 @@
+            _operator
+        );
+        require(
+            createdAt == 0 && operator.owner == address(0),
+            "Operator is already in use"
+        );
+        require(
+            _amount > minTStakeAmount,
+            "Amount to stake must be greater than minimum"
+        );
+        operator.owner = msg.sender;
+        operator.authorizer = _authorizer != address(0)
+            ? _authorizer
+            : msg.sender;
+        operator.beneficiary = _beneficiary != address(0)
+            ? _beneficiary
+            : payable(msg.sender);
+
+        operator.tStake = _amount;
+        /* solhint-disable-next-line not-rely-on-time */
+        operator.startTStakingTimestamp = block.timestamp;
+
+        emit TStaked(msg.sender, _operator);
+        emit OperatorStaked(
+            _operator,
+            operator.beneficiary,
+            operator.authorizer,
+            _amount
+        );
+        token.safeTransferFrom(msg.sender, address(this), _amount);
+    }
+
+    /// @notice Copies delegation from the legacy KEEP staking contract to T
+    ///         staking contract. No tokens are transferred. Caches the active
+    ///         stake amount from KEEP staking contract. Can be called by
+    ///         anyone.
+    function stakeKeep(address _operator) external override {
+        require(_operator != address(0), "Operator must be specified");
+        OperatorInfo storage operator = operators[_operator];
+
+        require(
+            operator.owner == address(0),
+            "Can't stake KEEP for this operator"
+        );
+
+        uint96 tAmount = getKeepAmountInT(_operator);
+
+        operator.keepStake = tAmount;
+        operator.owner = keepStakingContract.ownerOf(_operator);
+        operator.authorizer = keepStakingContract.authorizerOf(_operator);
+        operator.beneficiary = keepStakingContract.beneficiaryOf(_operator);
+        emit KeepStaked(operator.owner, _operator);
+        emit OperatorStaked(
+            _operator,
+            operator.beneficiary,
+            operator.authorizer,
+            tAmount
+        );
+    }
+
+    /// @notice Copies delegation from the legacy NU staking contract to T
+    ///         staking contract, additionally appointing beneficiary and
+    ///         authorizer roles. Caches the amount staked in NU staking
+    ///         contract. Can be called only by the original delegation owner.
+    function stakeNu(
+        address _operator,
+        address payable _beneficiary,
+        address _authorizer
+    ) external override {
+        require(_operator != address(0), "Operator must be specified");
+        OperatorInfo storage operator = operators[_operator];
+        (, uint256 createdAt, ) = keepStakingContract.getDelegationInfo(
+            _operator
+        );
+        require(
+            createdAt == 0 && operator.owner == address(0),
+            "Operator is already in use"
+        );
```

Comment:
For Keep version we do only:
```
        require(
            operator.owner == address(0),
            "Can't stake KEEP for this operator"
        );
```

which I *think* is sufficient. Why do we check `createdAt == 0` as well?